### DISCUSSION TITLE
Infrastructure Cost Estimate

### DISCUSSION THREAD

**Discussion by dysbulic** [MetaFam/TheGame#922] [General]
We are in the process of launching a Ceramic / IPFS mainnet node (#918), and to do so on Kubernetes @polimyl estimates will run us ~$70 / month.

I'd like to come up with an overarching infrastructure cost estimate based on the services we're offering.

Current deployed services are:

* [MyMeta](//my.metagame.wtf) running on [Render](//render.com) with test instances for pull requests generated on [Vercel](//vercel.com) & [Google Cloud](//cloud.google.com).
  * Render is running [staging](//test.metagame.wtf) and [production](//my.metagame.wtf) Hasura, PostgreSQL, & Next.js servers, and a Discord bot at $7 apiece, so $49 / month.
  * Vercel is generating test deployments of the Next.js server at $20 per month.
  * Google Cloud is hosting test instances of both the front and backends using Cloud Run at a cost of ~$15 / month.
* [The Wiki](//wiki.metagame.wtf) is running on Vercel and included in the $20 / month above.
* [SEED Claims](//claim.metagame.wtf) is running on Netlify for free.
* [The XP Explorer](//xp.metagame.wtf) is hosted on GitHub for free.
* [Discourse](//forum.metagame.wtf) is hosted on DigitalOcean on MetaCartel’s credits.

Upcoming services to be deployed are:

* A [Ceramic](//ceramic.network) / [IPFS](//ipfs.io) mainnet node at ~$70 for Kubernetes on Google Cloud, or $20 for a Virtual Private Server.
* An instance of the [IdentityLink](https://github.com/ceramicstudio/identitylink-services) serverless functions for account linking written to be deployed on Amazon's Web Services at an unknown cost.

So we're at ~$85 / month currently, and looking to roughly double that.

**Comment by polimyl** [comment]
hey, the first ever discussion in here! :tada: 
wouldn't such stuff usually be discussed in the forum though?

**Reply by dysbulic** [reply]
> wouldn't such stuff usually be discussed in the forum though?

I'd rather keep it all in GitHub for reference. Many people, like myself, check Discourse once a month or so. I don't expect there to be alot of discussion, it's mostly just a place for people to read a prepared document.

**Reply by polimyl** [reply]
i bet most non-devs are more active on discourse and check github once a month or so :P
and since this is the first ever discussion in here: how should other people find out about it?

**Reply by dysbulic** [reply]
> how should other people find out about it?

When we've got a reasonably sound number, I'll post a link in #-voting.

Really, only @META-DREAMER can fill in the blanks on the missing costs b/c they're almost certainly the one footing the bill. Unless @peth-yursick has a hand in it, but I doubt it.

**Comment by dysbulic** [comment]
If we we're to go the VPS route for hosting the Ceramic node, [Oracle's free tier](https://www.oracle.com/cloud/free/) includes “Two AMD Compute VMs”.

**Reply by polimyl** [reply]
gcp also includes a free vps, albeit with less resources than oracle cloud.
i also heard quite some bad stuff about oracle cloud and personally want to stay away from it because of that

**Comment by META-DREAMER** [comment]
> SEED Claims is running on Netlify at an unknown cost.

It's running for free currently

> Discourse is hosted on DigitalOcean at an unknown cost.

MetaCartel got digital ocean credits so this is also running for free currently
